app/src/main/java/com/h/pixeldroid/FeedActivity.kt

```diff
@@ -0,0 +1,64 @@
+package com.h.pixeldroid
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.TextView
+import androidx.fragment.app.FragmentManager
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.h.pixeldroid.api.PixelfedAPI
+import com.h.pixeldroid.models.Post
+import com.h.pixeldroid.models.Post.Companion.POST_FRAG_TAG
+import com.h.pixeldroid.objects.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class FeedActivity : AppCompatActivity() {
+    lateinit var feed : RecyclerView
+    lateinit var adapter : FeedRecyclerViewAdapter
+    var posts : List<Post> = ArrayList()
+
+    fun setContent(newPosts : ArrayList<Post>) {
+        feed = findViewById(R.id.feedList)
+        feed?.setHasFixedSize(true)
+        feed?.layoutManager = LinearLayoutManager(this)
+        posts = newPosts
+        adapter = FeedRecyclerViewAdapter(context = this)
+        feed?.adapter = adapter
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_feed)
+
+        var statuses: ArrayList<Status>? = null
+        val BASE_URL = "https://pixelfed.de/"
+
+        val pixelfedAPI = PixelfedAPI.create(BASE_URL)
+
+        val newPosts = ArrayList<Post>()
+
+        pixelfedAPI.timelinePublic(null, null, null, null, null)
```

Comment:
No need for null arguments since they are already set as the default values in the function declaration.